pkgs/tools/package-management/nix-prefetch/default.nix

```diff
@@ -1,67 +1,74 @@
-{ stdenv, fetchFromGitHub, makeWrapper
-, asciidoc, docbook_xml_dtd_45, docbook_xsl, libxml2, libxslt
-, coreutils, gawk, gnugrep, gnused, jq, nix }:
+{ stdenv, fetchFromGitHub, installShellFiles, makeWrapper, asciidoc
+, docbook_xml_dtd_45, git, docbook_xsl, libxml2, libxslt, coreutils, gawk
+, gnugrep, gnused, jq, nix }:
 
-stdenv.mkDerivation rec {
+let
+  binPath = stdenv.lib.makeBinPath [ coreutils gawk git gnugrep gnused jq nix ];
+
+in stdenv.mkDerivation rec {
   pname = "nix-prefetch";
-  version = "0.3.0";
+  version = "0.3.1";
 
   src = fetchFromGitHub {
     owner = "msteen";
     repo = "nix-prefetch";
     rev = version;
-    sha256 = "0b9gdi7xzmfq0j258x724xsll8gi31m0m4pzfjkqinlm6zwr3sgm";
+    sha256 = "15h6f743nn6sdq8l771sjxh92cyzqznkcs7szrc7nm066xvx8rd4";
+    # the stat call has to be in a subshell or we get the current date
+    extraPostFetch = ''
+      echo $(stat -c %Y $out) > $out/.timestamp
+    '';
   };
 
+  postPatch = ''
+    lib=$out/lib/${pname}
+
+    substituteInPlace doc/nix-prefetch.1.asciidoc \
+      --subst-var-by version $version
+
+    substituteInPlace src/main.sh \
+      --subst-var-by lib $lib \
+      --subst-var-by version $version
+
+    substituteInPlace src/tests.sh \
+      --subst-var-by bin $out/bin
+  '';
+
   nativeBuildInputs = [
+    asciidoc
+    docbook_xml_dtd_45
+    docbook_xsl
+    installShellFiles
+    libxml2
+    libxslt
     makeWrapper
-    asciidoc docbook_xml_dtd_45 docbook_xsl libxml2 libxslt
   ];
 
-  configurePhase = ''
-    . configure.sh
-  '';
+  dontConfigure = true;
 
   buildPhase = ''
-    a2x -f manpage doc/nix-prefetch.1.asciidoc
+    a2x -a revdate=$(date --date=@$(cat $src/.timestamp) +%d/%m/%Y) \
```

Comment:
Not sure if Nix already does this for us, but do we need to add `--utc` to make it more pure?